src/plugins/data/public/query/query_string/language_service/lib/query_result.tsx

```diff
@@ -28,21 +28,66 @@ export interface QueryStatus {
     statusCode?: number;
   };
   elapsedMs?: number;
+  startTime?: number;
 }
 
+// This is the time in milliseconds that the query will wait before showing the loading spinner
+const BUFFER_TIME = 3000;
+
 export function QueryResult(props: { queryStatus: QueryStatus }) {
   const [isPopoverOpen, setPopover] = useState(false);
+  const [elapsedTime, setElapsedTime] = useState(0);
   const onButtonClick = () => {
     setPopover(!isPopoverOpen);
   };
 
+  const getTime = () => {
+    const time = new Date().getTime() - props.queryStatus.startTime!;
+    if (time > BUFFER_TIME) {
+      setElapsedTime(time);
+    } else {
+      setElapsedTime(0);
+    }
+  };
+
+  useEffect(() => {
+    const interval = setInterval(getTime, 1000);
+
+    return () => clearInterval(interval);
+  });
+
+  if (props.queryStatus.status === ResultStatus.LOADING) {
+    if (elapsedTime < BUFFER_TIME) {
+      return null;
+    }
+    const time = Math.floor(elapsedTime / 1000);
+    return (
+      <EuiButtonEmpty
+        color="text"
+        size="xs"
+        onClick={() => {}}
+        isLoading
+        data-test-subj="queryResultLoading"
+      >
+        {`Loading ${time} s`}
+      </EuiButtonEmpty>
+    );
+  }
+
   if (props.queryStatus.status === ResultStatus.READY) {
+    let message;
+    if (!props.queryStatus.elapsedMs) {
+      message = 'Completed';
+    } else if (props.queryStatus.elapsedMs < 1000) {
+      message = `Completed in ${props.queryStatus.elapsedMs} ms`;
+    } else {
+      message = `Completed in ${Math.floor(props.queryStatus.elapsedMs / 1000)} s`;
```

Comment:
Not sure what the UX is but as a user interested it knowing how long it took, flooring to the second is not good enough.

```suggestion
      message = `Completed in ${(props.queryStatus.elapsedMs / 1000).toFixed(1)} s`;
```